### PR TITLE
Product path query

### PR DESCRIPTION
**Pull request by piotr-zyskowski-rai**

### CODE REVIEW

**Comment by michalpelka** [comment]
This PR should also cover this case:
https://github.com/RobotecAI/o3de-ros2-gem/issues/165

File: Code/Source/RobotImporter/URDF/UrdfParser.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@ namespace ROS2
 {
     urdf::ModelInterfaceSharedPtr UrdfParser::Parse(const AZStd::string& xmlString)
     {
+        std::setlocale(LC_NUMERIC, "en_US.UTF-8");
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/UrdfParser.cpp]: Unrelated change from another PR

File: Code/Source/RobotImporter/URDF/CollidersMaker.cpp [hunk 2]
```diff
@@ -24,6 +25,86 @@
...
+        AZ::IO::Path GetFullURDFMeshPath(AZ::IO::Path modelPath, AZ::IO::Path meshPath)
+        {
+            modelPath.RemoveFilename();
+            AZ::StringFunc::Replace(meshPath.Native(), "package://", "", true, true);
+            modelPath /= meshPath;
+
+            return modelPath;
+        }
+
+        AZStd::optional<AZ::IO::Path> GetMeshProductPathFromSourcePath(const AZ::IO::Path& sourcePath)
```
Comment (adamdbrw) [inline Code/Source/RobotImporter/URDF/CollidersMaker.cpp:40]: This looks like a Utils function - not related to Colliders. Best to separate it out.
